### PR TITLE
Added AttributesPreprocessor

### PR DESCRIPTION
**Pull request by maxpiva**
if enabled it will call the preprocessor to obtain the serializable property attributes instead the normal reflection method, this enable to dynamically add/change/remove attributes to the serializables properties in run-time.

### CODE REVIEW

File: YAXLib/YAXSerializer.cs [hunk 1]
```diff
@@ -228,6 +228,11 @@ public YAXSerializationOptions SerializationOption
         }
 
         /// <summary>
+        /// Gets or Sets the Attribute Preprocessor callback
+        /// </summary>
+        public IAttributesPreprocessor AttributesPreprocessor { get; set; } = null;
```
Comment (sinairv) [inline YAXLib/YAXSerializer.cs]: YAXLib is still in C# 5.0 era, this seems to be what's breaking the build. BTW default value for any reference type is always `null`.

File: YAXLib/MemberWrapper.cs [hunk 2]
```diff
@@ -125,18 +125,22 @@ public MemberWrapper(MemberInfo memberInfo, YAXSerializer callerSerializer)
 
             TreatErrorsAs = callerSerializer != null ? callerSerializer.DefaultExceptionType : YAXExceptionTypes.Error;
 
+            Attribute[] attributes;
+            attributes = callerSerializer?.AttributesPreprocessor != null ? callerSerializer?.AttributesPreprocessor.PreProcessAttributes(m_memberInfo) : Attribute.GetCustomAttributes(m_memberInfo, true);
```
Comment (sinairv) [inline YAXLib/MemberWrapper.cs]: line too long. Please break down to:

```
attributes = callerSerializer?.AttributesPreprocessor != null 
    ? callerSerializer?.AttributesPreprocessor.PreProcessAttributes(m_memberInfo) 
    : Attribute.GetCustomAttributes(m_memberInfo, true);
```

**Comment by sinairv** [comment]
I don't see any unit tests supporting the reason behind this change.

**Comment by maxpiva** [comment]
Will create a unit test and remove C# syntax.

**Comment by maxpiva** [comment]
Sorry there was a double check in the first foreach loop

**Pull request closed without merging**